tests/framework/hypervisor/device.go

```diff
@@ -20,26 +20,16 @@
 package hypervisor
 
 import (
-	k8sv1 "k8s.io/api/core/v1"
 	"kubevirt.io/client-go/kubecli"
 
-	"kubevirt.io/kubevirt/pkg/virt-config/featuregate"
-	"kubevirt.io/kubevirt/pkg/virt-controller/services"
+	"kubevirt.io/kubevirt/pkg/hypervisor"
 	"kubevirt.io/kubevirt/tests/libkubevirt"
 )
 
-// GetDevice returns the appropriate hypervisor device resource name
-// based on the current KubeVirt configuration. If HyperVLayered feature gate
-// is enabled, it returns HyperVDevice, otherwise KvmDevice.
-func GetDevice(virtClient kubecli.KubevirtClient) k8sv1.ResourceName {
+// GetDevice returns the appropriate hypervisor device name
+// based on the current KubeVirt configuration.
+func GetDevice(virtClient kubecli.KubevirtClient) string {
 	kv := libkubevirt.GetCurrentKv(virtClient)
-	if kv.Spec.Configuration.DeveloperConfiguration != nil {
-		featureGates := kv.Spec.Configuration.DeveloperConfiguration.FeatureGates
-		for _, fg := range featureGates {
-			if fg == featuregate.HyperVLayered {
-				return services.HyperVDevice
-			}
-		}
-	}
-	return services.KvmDevice
+	hypervisorName := kv.Spec.Configuration.HypervisorConfiguration.Name
+	return hypervisor.NewHypervisor(hypervisorName).GetDevice()
```

Comment:
Can we just `return kv.Spec.Configuration.HypervisorConfiguration.Name`

Reply:
hypervisor name (hyperv-layered) is different than name of device (mshv)